### PR TITLE
Mk/rebroadcast detection refactor

### PR DESCRIPTION
**Pull request by maciejka**

### CODE REVIEW

File: src/blockchain/transactions.ts [hunk 1]
```diff
@@ -103,6 +103,38 @@ export type TxState = {
 
 let txCounter: number = 1;
 
+type NodeCallback<I, R> = (
+  i: I,
```
Comment (peculiarity) [inline src/blockchain/transactions.ts]: What does this `i` should represent?

Comment (peculiarity) [reply src/blockchain/transactions.ts]: I see it depends on the type provided but still it represents something

File: src/blockchain/transactions.ts [hunk 2]
```diff
@@ -103,6 +103,38 @@ export type TxState = {
...
+  callback: (err: any, r: R) => any,
+) => any;
+
+type GetTransactionReceipt = NodeCallback<string, { blockNumber: number, transactionHash: string }>;
+
+interface TransactionLike {
```
Comment (peculiarity) [inline src/blockchain/transactions.ts]: `TransactionPartial` | `PartialTransaction` probably ?

File: src/blockchain/transactions.ts (hunk 1 above)
Comment (maciejka) [reply src/blockchain/transactions.ts]: type of function that fits node callback definition

**Pull request merged**